src/feesEngine.ts

```diff
@@ -0,0 +1,98 @@
+import { earnCall, EarnInfo } from './partners/earn'
+import { ethGasStationCall, GasInfo } from './partners/ethGasStation'
+import { BitcoinInfo, mempoolSpaceCall } from './partners/mempoolSpace'
+import { config } from './utils/config'
+import { normalizeDate } from './utils/utils'
+const nano = require('nano')
+const promisify = require('promisify-node')
+
+const LOOP_TIME_MS = config.timeBetweenCyclesInMinutes * 60 * 1000
+const mylog = console.log
+
+// Nano for CouchDB
+// =============================================================================
+const nanoDb = nano(config.dbFullpath)
+const dbEthGS = nanoDb.db.use('fees_ethgasstation')
+const dbEarn = nanoDb.db.use('fees_mempoolspace')
+const dbMempool = nanoDb.db.use('fees_earn')
+promisify(dbEthGS)
+promisify(dbEarn)
+promisify(dbMempool)
+
+const snooze = async (ms: number): Promise<void> =>
+  await new Promise((resolve: Function) => setTimeout(resolve, ms))
+
+const feesLoop = async (): Promise<GasInfo | BitcoinInfo | EarnInfo> => {
+  while (true) {
+    let preppedData
+    let prePrepData
+    const earn = earnCall()
+    const mempool = mempoolSpaceCall()
+    const ethGasStation = ethGasStationCall()
```

Comment:
Handle result/rejected promises on each one and not on returned array from promise.all